spring-cloud-starter-stream-processor-image-recognition/src/main/java/org/springframework/cloud/stream/app/image/recognition/processor/ImageRecognitionOutputMessageBuilder.java

```diff
@@ -0,0 +1,124 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.image.recognition.processor;
+
+import java.awt.*;
+import java.awt.geom.Rectangle2D;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.imageio.ImageIO;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.cloud.stream.app.tensorflow.processor.DefaultOutputMessageBuilder;
+import org.springframework.cloud.stream.app.tensorflow.processor.TensorflowCommonProcessorProperties;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.tuple.JsonStringToTupleConverter;
+import org.springframework.tuple.Tuple;
+
+/**
+ * Extends the {@link DefaultOutputMessageBuilder} with ability to to augment the input image with the
+ * recognized labels.
+ *
+ * @author Christian Tzolov
+ */
+public class ImageRecognitionOutputMessageBuilder extends DefaultOutputMessageBuilder {
+
+	private static final Log logger = LogFactory.getLog(ImageRecognitionOutputMessageBuilder.class);
+
+	public static final String IMAGE_FORMAT = "jpg";
+
+	private final Color textColor = Color.BLACK;
+	private final Color bgColor = new Color(167, 252, 0);
+	private final boolean drawLabels;
+
+	public ImageRecognitionOutputMessageBuilder(boolean drawLabels,
+			TensorflowCommonProcessorProperties properties) {
+		super(properties);
+		this.drawLabels = drawLabels;
+	}
+
+	@Override
+	public MessageBuilder<?> createOutputMessageBuilder(Message<?> inputMessage, Object computedScore) {
+		Message<?> annotatedInput = inputMessage;
+
+		if (this.drawLabels) {
+			byte[] annotatedImage = drawLabels((byte[]) inputMessage.getPayload(), computedScore);
+			annotatedInput = MessageBuilder.withPayload(annotatedImage).build();
+		}
+
+		return super.createOutputMessageBuilder(annotatedInput, computedScore);
+	}
+
+	/**
+	 * Augment the input image by adding the recognized classes.
+	 *
+	 * @param imageBytes input image as byte array
+	 * @param result computed recognition labels
+	 * @return the image augmented with recognized labels.
+	 */
+	private byte[] drawLabels(byte[] imageBytes, Object result) {
+		try {
+			if (result != null) {
+				BufferedImage originalImage = ImageIO.read(new ByteArrayInputStream(imageBytes));
+
+				Graphics2D g = originalImage.createGraphics();
+				g.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON);
+				FontMetrics fm = g.getFontMetrics();
+
+				Tuple resultTuple = new JsonStringToTupleConverter().convert(result.toString());
+				ArrayList<Tuple> labels = (ArrayList) resultTuple.getValues().get(0);
```

Comment:
![image](https://user-images.githubusercontent.com/1071590/37002907-369095fa-2099-11e8-9c21-c8ff5246c8ae.png)


Would be better to have generic args and use `@SuppressWarning` on the method.

Reply:
+1. I've replaced with `ArrayList<Tuple> labels  = resultTuple.getValue("labels", ArrayList.class);`